testing/opae-c/test_enum_c.cpp

```diff
@@ -316,9 +329,19 @@ TEST_P(enum_c_p, device_id) {
   EXPECT_EQ(
       fpgaEnumerate(&filter_, 1, tokens_.data(), tokens_.size(), &num_matches_),
       FPGA_OK);
-  EXPECT_EQ(num_matches_, GetNumDeviceID() * 2);
+  EXPECT_EQ(num_matches_, GetNumDeviceID() * (device.num_vfs == 0 ? 2 : 1));
 
-  DestroyTokens();
+  num_matches_ = 0;
+  for (int i = 1; i < device.num_vfs+1; ++i) {
+    ASSERT_EQ(fpgaPropertiesSetDeviceID(filter_, device.device_id+i), FPGA_OK);
+    EXPECT_EQ(
+        fpgaEnumerate(&filter_, 1, tokens_.data(), tokens_.size(), &num_matches_),
```

Comment:
Isn't this loop bleeding tokens?  ie, each time through tokens_.data() is set to a cloned token.  Each iteration after the first will leak the token.

Reply:
good catch 😄

Reply:
miss this one?